src/app/server/OnboardingCodesUtil.cpp

```diff
@@ -37,29 +37,29 @@ using namespace ::chip::DeviceLayer;
 
 void PrintOnboardingCodes(chip::RendezvousInformationFlags aRendezvousFlags)
 {
-    chip::SetupPayload payload;
+    chip::PayloadContents payload;
 
-    CHIP_ERROR err = GetSetupPayload(payload, aRendezvousFlags);
+    CHIP_ERROR err = GetPayloadContents(payload, aRendezvousFlags);
     if (err != CHIP_NO_ERROR)
     {
-        ChipLogError(AppServer, "GetSetupPayload() failed: %s", chip::ErrorStr(err));
+        ChipLogError(AppServer, "GetPayloadContents() failed: %s", chip::ErrorStr(err));
     }
 
     PrintOnboardingCodes(payload);
 }
 
-void PrintOnboardingCodes(const chip::SetupPayload & payload)
+void PrintOnboardingCodes(const chip::PayloadContents & payload)
 {
-    std::string qrCode;
-    std::string manualPairingCode;
+    char payloadBuffer[chip::QRCodeBasicSetupPayloadGenerator::kMaxQRCodeBase38RepresentationLength];
+    chip::MutableCharSpan qrCode(payloadBuffer);
```

Comment:
No, this is going to be buggy.

GetQRCode, if you dig way down into it, will do the following:
1. Try to put a null-terminated string into the provided MutableCharSpan.  This means that if the actual code is in fact `kMaxQRCodeBase38RepresentationLength` long it will not fit into this buffer.
2. Not resize it to the actual size of the string, since it's meant to be null-terminated.  Which means when you pass this thing to `GetQRCodeUrl` and it uses `size()` it will get a bogus value and read random data....

We need to make sure that we are doing the right things here.  Ideally we should stop doing the null-termination thing with CharSpans, adjust their sizes to match their data in the callees, and any place where we are currently getting data out of those charspans and relying on them to be null-terminated we fix that....  Because having these CharSpans around whose size includes random data that's not actually part of the data we put into the span is just a footgun.

Reply:
Yeah, I was basing on the existing implementation: https://github.com/project-chip/connectedhomeip/blob/master/src/controller/CommissioningWindowOpener.cpp#L147, but you are completely right, that may cause problems. I tried to follow your suggestion, so here is what I changed:
* Added reducing the span size in `base38Encode` and `payloadDecimalStringRepresentation`methods by 1 to not include null termination character.
* Increased all buffers that are used to get QR code or manual pairing code by 1 to include null terminator.
The overall I wanted to make the data be ended with null termination character, but to not include it in the span size.